cegui/src/XMLParserModules/TinyXML/XMLParser.cpp

```diff
@@ -64,45 +64,30 @@ namespace CEGUI
         d_handler = &handler;
 
         // Create a buffer with extra bytes for a newline and a terminating null
-        size_t size = source.getSize();
-        char* buf = new char[size + 2];
-        memcpy(buf, source.getDataPtr(), size);
+        const size_t size = source.getSize();
+        std::string buf;
+        buf.reserve(size + 1);
+        buf.assign(source.getDataPtr(), size);
         // PDT: The addition of the newline is a kludge to resolve an issue
         // whereby parse returns 0 if the xml file has no newline at the end but
         // is otherwise well formed.
-        buf[size] = '\n';
-        buf[size+1] = 0;
+        buf += '\n';
 
         // Parse the document
         TiXmlDocument doc;
-        if (!doc.Parse(static_cast<const char*>(buf)))
+        if (!doc.Parse(buf.data()))
         {
-            // error detected, cleanup out buffers
-            delete[] buf;
-
-            // throw exception
+            // error detected, throw exception
             CEGUI_THROW(FileIOException("an error occurred while "
                 "parsing the XML document - check it for potential errors!."));
         }
 
         const TiXmlElement* currElement = doc.RootElement();
         if (currElement)
         {
-            CEGUI_TRY
-            {
-                // function called recursively to parse xml data
-                processElement(currElement);
-            }
-            CEGUI_CATCH(...)
-            {
-                delete [] buf;
-
-                CEGUI_RETHROW;
-            }
+           // function called recursively to parse xml data
+           processElement(currElement);
```

Comment:
`processElement` may throw and seems like you've left it out of a try-catch block.

Reply:
No try/catch block needed when `buf` is a std::string. It will always handle the cleanup on exceptions and early returns